src/transform/common.jl

```diff
@@ -65,11 +65,11 @@ end
 
 function cache_dae_ci!(old_ci, src, debuginfo, abi, owner)
     mi = old_ci.def
-    edges = Core.svec(mi)
+    edges = Core.svec(old_ci)
     daef_ci = CodeInstance(abi === nothing ? old_ci.def : Core.ABIOverride(abi, old_ci.def), owner, Tuple, Union{}, nothing, src, Int32(0),
-        UInt(1)#=ci.min_world=#, old_ci.max_world, old_ci.ipo_purity_bits,
+        old_ci.min_world, old_ci.max_world, old_ci.ipo_purity_bits,
         nothing, debuginfo, edges)
-    ccall(:jl_method_instance_add_backedge, Cvoid, (Any, Any, Any), mi, nothing, daef_ci)
+    Compiler.store_backedges(daef_ci, edges)
```

Comment:
`store_backedges`  will [ignore](https://github.com/JuliaLang/julia/blob/2a5cfee2807add7babf00ddd93da505e2cce5719/Compiler/src/typeinfer.jl#L752) a few `CodeInstance`s that we generate, that's why I used `jl_method_instance_add_backedge`. We might want to remove this check in Compiler instead, but I don't know what the implications are/what were the motivations to have it in the first place.
```suggestion
    ccall(:jl_method_instance_add_backedge, Cvoid, (Any, Any, Any), old_ci, nothing, daef_ci)
```

Reply:
(this should be picked up by tests)

Reply:
I don't think we're using any CodeInstances that don't have an associated method.

Reply:
I don't recall exactly what we had, but this check did trigger for some. That's most likely what [broke the tests](https://github.com/CedarEDA/DAECompiler.jl/actions/runs/14982833801/job/42090720227?pr=32#step:6:850).

Reply:
It's possible that check in base needs to use a `get_ci_mi(ci).def` instead so that our ABI override code instances work properly, but I didn't think we put those into `edges`

Reply:
The check is for callers not edges, which are the ABI override code instances.

Reply:
Ah. Yes, submit a PR to base to put in the get_ci_mi then

Reply:
PR: https://github.com/JuliaLang/julia/pull/58394
I confirmed that this change allows `store_backedges` to work correctly with our `CodeInstance`s, instead of needing to call `jl_method_instance_add_backedge` manually.